task_1/task_1.py

```diff
@@ -0,0 +1,302 @@
+        self.headers = {
+            "accept": "application/json",
+            "Authorization": "Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIzMTI3NGFmYTRlNTUyMjRjYzRlN2Q0NmNlMTNk"
+                             "OTZjOSIsInN1YiI6IjVkNmZhMWZmNzdjMDFmMDAxMDU5NzQ4OSIsInNjb3BlcyI6WyJhcGlfcmVhZ"
+                             "CJdLCJ2ZXJzaW9uIjoxfQ.lbpgyXlOXwrbY0mUmP-zQpNAMCw_h-oaudAJB6Cn5c8"
+        }
+        self.full_data = []
+        self.filtered_list = []
+        self.top_list_from_genre = []
+        self.collection_of_present_genres = ()
+        self.collection_movie_grouped_by_genre = ()
+        self.restructured_movies_list = []
+        self.csv_out_file_name_fulldata = ''
+        self.csv_out_file_name_fulldata_default = 'out_full_data.csv'
+        self.csv_out_file_name_each_4 = ''
+        self.csv_out_file_name_each_4_default = 'out_each_4_move.csv'
+
+        self.__request()
+
+    @staticmethod
+    def __input_numder_page():
+        i = 0
+        while i < 10:
+            user_input = input('How many page download(max 500): ')
+            try:
+                n = int(user_input)
+            except ValueError:
+                print("Bad input.\nPlease enter a valid integer.")
+                i += 1
+            else:
+                return n
+        print('Too many input errors\n------ Good by. ------')
+        exit()
+
+    def __request(self):
+        # comeback count of need pages (self.page_count)
+        self.full_data.clear()
+        i = 1
+        while i <= self.page_count:
+            response = requests.get(url=self.url.format(i), headers=self.headers)
+            try:
+                response.raise_for_status()
+            except requests.exceptions.RequestException as e:
+                print("An error occurred:", e)
+                if response.status_code == 422:
+                    print(f'Error: page number {i} not present.')
+                elif response.status_code == 429:
+                    print(
+                        f'response=429\nYour request count ({i}) '
+                        'is over the allowed limit of (40) per second.\nPlease wait 10s.')
+                    time.sleep(10)
+                    continue
+                elif response.status_code == 404:
+                    print('Error HTTP: Page not found.')
+                break
+            else:
+                print(i)
+                i += 1
+                for film in response.json()["results"]:
+                    self.full_data.append(film)
+
+    def validate_path(self, f_path):
+        file_path = os.path.dirname(f_path)
+        file_name = os.path.basename(f_path)
+
+        if file_path:
+            if not os.path.exists(file_path):
+                os.makedirs(file_path)
+            file_path += '\\'
+
+        return file_path, file_name
+
+    def __add_file_extention(self, name):
+        f_name = name if (name.find('.csv', -4)) != -1 else name + '.csv'
```

Comment:
controversial 
you could use .endwith() method
and what will you do if someone decides to pass you file with extension .txt?